### PR TITLE
Firehose metrics add subsystem, create override_default_tags

### PR DESCRIPTION
**Pull request by ryantanjunming**
# Description

- Added subsystem value to common attributes of firehose metrics
- Added override_default_tags to allow users to override the default tags we set

# How Has This Been Tested?

Local testing with our AWS account

# Checklist:
- [ ] I have updated the relevant example in the examples directory for the module.
- [ ] I have updated the relevant test file for the module.
- [ ] This change does not affect module (e.g. it's readme file change)

### CODE REVIEW

**Comment by macmiranda** [comment]
Hey @ryantanjunming just an idea, perhaps `var.override_default_tags` should be a boolean. `var.user_supplied_tags` remains the variable for user input. The former just changes the behavior of the merge. If `true`, the user-supplied tags come later in the merge, if `false` the other way around (which is the default). 

Having two maps for user-supplied tags will be confusing.

**Comment by ryantanjunming** [comment]
@macmiranda i'll set it as `var.use_default_tags` boolean default to true. If false, it will remove the default tags and leave just the var.user_supplied_tags.

So it's possible to have it empty also.

File: modules/firehose/README.md [hunk 1]
```diff
@@ -218,6 +218,7 @@ then the CloudWatch metric stream must be configured with the same format, confi
 | <a name="input_additional_metric_statistics_enable"></a> [input\_additional\_metric\_statistics\_enable](#input\_additional\_metric\_statistics\_enable) | To enable the inclusion of additional statistics to the streaming metrics | `bool` | `true` | no |
 | <a name="input_additional_metric_statistics"></a> [input\_additional\_metric\_statistics](#input\_additional\_metric\_statistics) | For each entry, specify one or more metrics (metric_name and namespace) and the list of additional statistics to stream for those metrics. Each configuration of metric name and namespace can have a list of additional_statistics included into the AWS CloudWatch Metric Stream. | `list(object({additional_statistics=list(string), metric_name=string, namespace=string}))` | See variables.tf | no |
 | <a name="input_user_supplied_tags"></a> [user_supplied_tags](#input\_user_supplied_tags) | Tags supplied by the user to populate to all generated resources | `map(string)` | n/a | no |
+| <a name="input_override_default_tags"></a> [override_default_tags](#input\_override_default_tags) | Override the default tags in the module by supplying a map of key-value pairs | `map(string)` | n/a | no |
```
Comment (macmiranda) [inline modules/firehose/README.md:221]: I think this needs to be updated

File: modules/firehose/variables.tf [hunk 2]
```diff
@@ -170,6 +170,12 @@ variable "user_supplied_tags" {
   default     = {}
 }
 
+variable "override_default_tags" {
+  description = "Override the default tags in the module by supplying a map of key-value pairs"
```
Comment (macmiranda) [inline modules/firehose/variables.tf:174]:
```suggestion
  description = "Whether or not to override the default tags in the module"
```

**Review by guyrenny — APPROVED** [review]

**Pull request merged**